pkg/operator/controller/nodeobservabilityrun/controller.go

```diff
@@ -89,92 +90,95 @@ func (r *NodeObservabilityRunReconciler) Reconcile(ctx context.Context, req ctrl
 		return
 	}
 
-	// interact with MCO only if we are executing CrioKubeletProfile runs
-	// also bypass if the runtype is e2e-test
-	if instance.Spec.RunType == nodeobservabilityv1alpha1.CrioKubeletProfile {
-		if err := r.ensureNOMC(ctx); err != nil {
-			return ctrl.Result{}, err
-		}
-		// this will ensure we dont trigger the agent call
-		// until the mco and mcp have been updated successfully
-
-		enabled, err := r.checkNOMCStatus(ctx, true)
-		if err != nil {
-			r.Log.Error(err, "NodeObservabilityMachineConfig enable status check failed")
-			return ctrl.Result{RequeueAfter: 1 * time.Minute}, err
-		}
-		if enabled {
-			r.Log.V(3).Info("NodeObservabilityMachineConfig CrioKubeletProfile enabled")
-		} else {
-			r.Log.Info("NodeObservabilityMachineConfig CrioKubeletProfile not enabled yet")
-			return ctrl.Result{RequeueAfter: 1 * time.Minute}, nil
-		}
-	}
-
-	// this code should be included in the 'if' statement above
-	// as it is specific to CrioKubeletProfiling
-	// once new profiling features are added it can be moved
-
-	// continue once machineconfigpool has been updated successfully
 	if finished(instance) {
 		r.Log.V(3).Info("Run for this instance has been completed already")
-		// check the CR if the RestoreMCOStateAfterRun flag is set
-		if instance.Spec.RestoreMCOStateAfterRun {
-			err := r.disableCrioKubeletProfile(ctx)
-			if err != nil {
-				return ctrl.Result{}, err
-			}
-
-			disabled, err := r.checkNOMCStatus(ctx, false)
-			if err != nil {
-				r.Log.Error(err, "NodeObservabilityMachineConfig disable status check failed")
-				return ctrl.Result{}, err
-			}
-			if disabled {
-				r.Log.Info("NodeObservabilityMachineConfig CrioKubeletProfile disabled")
-				nomc := r.desiredMCO()
-				if err := r.deleteNOMC(ctx, nomc); err != nil {
-					if !errors.IsNotFound(err) {
-						return ctrl.Result{}, err
-					}
-				}
-			} else {
-				r.Log.Info("NodeObservabilityMachineConfig CrioKubeletProfile not disabled yet")
-				return ctrl.Result{RequeueAfter: 1 * time.Minute}, nil
-			}
-		}
 		return
 	}
 
 	defer func() {
-		errUpdate := r.Status().Update(ctx, instance, &client.UpdateOptions{})
+		errUpdate := r.updateStatus(ctx, instance)
 		if errUpdate != nil {
-			r.Log.Error(err, "failed to update status")
+			r.Log.Error(errUpdate, "failed to update status")
```

Comment:
The error returned from the reconciler will be logged anyway, so you'll have it twice.